pages/api/calendar.ts

```diff
@@ -0,0 +1,113 @@
+import fs from 'fs/promises'
+import path from 'path'
+import { calendar_v3, google } from 'googleapis'
+import { CalendarEvent } from '../../models/event'
+import { NextApiRequest, NextApiResponse } from 'next/types'
+
+// If modifying these scopes, update the service account permissions.
+const SCOPES = ['https://www.googleapis.com/auth/calendar.readonly']
+const CREDENTIALS_PATH = path.join(process.cwd(), 'credentials.json')
+/**
+ * Loads the service account credentials from the JSON file.
+ *
+ * @return {Promise<Object>}
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+async function loadServiceAccountCredentials(): Promise<any> {
+  const content = await fs.readFile(CREDENTIALS_PATH)
+
+  return JSON.parse(content.toString())
+}
```

Comment:
If I remember correctly the Google module should be able to load the credentials implicitly with an environmental variable.

Reply:
Google does this on their own quickstart guide: https://developers.google.com/calendar/api/quickstart/nodejs

Reply:
Could probably make deployment simpler if all configurable vars were in one env file, something like this seems to work
https://cloud.google.com/nodejs/docs/reference/google-auth-library/latest#loading-credentials-from-environment-variables